### ISSUE TITLE
No transaltion for nested or child components 

### ISSUE THREAD

**Issue by houssemFat** [intlify/vue-i18n-next#106]
Hello, thank you for your work. I'm trying to use the `vue-i18n-next` with `vue 3` but it does not seem to work for nested components (or child component). 

### vue & vue-i18n version
- vue: 3.0.0-beta.1
- vue-i18n: 9.0.0-alpha.16

### Steps to reproduce 
I tried codesandbox.io but there are some issues with vue 3 import, so here is the complete list of files to reproduce the issue.

In the App wrapper view, there is a i18n text which works and an other component `Login` which also use i18n. But i18n only works for the wrapper container and not for the login view.

##### File `App.vue`
```
<template>
    {{t("login.title")}}
    <br/>
    does it works for login ?
    <Login></Login>
</template>

<script>
        import {useI18n} from "vue-i18n";
	import Login from "./components/Accounts/Login";

	export default {
		name: 'App',
		components: {
			Login
		},
		setup() {
			return useI18n()
		}
	}
</script>

<style>

</style>

```
##### File `Login.vue`
```
<template>
<div>
    {{ t("login.title") }}
</div>
</template>

<script>

	export default {
		name: 'Login',
		components: {
		}
	}
</script>

<style lang="scss">
</style>

```
##### File `main.js`
```
import {createApp} from 'vue'
import App from './App.vue'
import router from "./routes/routes";
import i18n from "./i18n";

const app = createApp(App);

app.use(router);
app.use(i18n);

app.mount('#app')

```

##### File `i18n.js`
```
import {createI18n} from 'vue-i18n'

const i18n = createI18n({
	locale: 'en',
	globalInstall: true,
	messages: {
		en: {'login.title': 'test'}
	}
})
export default i18n;

```

### What is Expected?
I expect to translate the key inside the login view also and nested children.
### What is actually happening?
Only the translation inside `App.vue` works, the translate for `Login.vue` does not work.
#### Screenshots
- UI : 
![image](https://user-images.githubusercontent.com/34309592/92459286-c807e280-f1c6-11ea-88fd-be737d684fbd.png)
- Console : 
![image](https://user-images.githubusercontent.com/34309592/92459334-d655fe80-f1c6-11ea-9af5-35ad3cbcc3c0.png)

**Comment by kazupon** [comment]
I've checked your code!
`Login.vue` need to run the `useI18n` in `setup`.
vue-i18n-next (v9) or later runs in `composable` mode, optimized for the composition API by default.

If you want similar behavior to vue-i18n@v8.x, you need to set `createI18n` to `legacy: true`.

**Comment by houssemFat-DDF** [comment]
Thank you for your quick reply. So starting from this version and later, i should run the` useI18n`  in the `setup` `fn` for every component. Is is repetitive knowing that the `i18n` can be loaded in the main app wrapper ?

**Comment by kazupon** [comment]
> i should run the` useI18n` in the `setup` `fn` for every component.

Yes, that's right.
Composition API style requires that you **explicitly** call `useI18n` and return of that function taht such as `t` or `d`, these need return as setup context.

>  Is is repetitive knowing that the i18n can be loaded in the main app wrapper ?

Sorry, I didn't understand. 
Could you be able to explain in a little more detail, please? 🙏

**Comment by houssemFat** [comment]
Sorry for the late response, what i meant, suppose that i have  a tree of an app like below, with the new api, i should import usei18n for all nested components.
```
App >>
           Account  >>
                              Login.vue  (import useI18n)
                              Password.vue  (import useI18n)
                              Register.vue  (import useI18n)
        Dashboard >>
                             Module.A.vue  (import useI18n)
                             Module.B.vue  (import useI18n)
.....
```
If i use i18n in all these views  why I can't import it only once in the main app container ?  
```
App   (import useI18n) >>
           Account  >>
                              Login.vue (use the instance created in the top container)
                              Password.vue  (use the instance created in the top container)
                              Register.vue  (use the instance created in the top container)
        Dashboard >>
                             Module.A.vue   (use the instance created in the top container)
                             Module.B.vue  (use the instance created in the top container)
...
```

**Comment by kazupon** [comment]
Thank you for your feedback!

You can use API with global scope that is generated when you run `createI18n` at the top of the application.
You need to run `useI18n` with **no options** in each component's `setup`, so the function returns a global `composer`. 
You can check the example:
https://github.com/intlify/vue-i18n-next/blob/master/examples/composable/scope/global.html

The `composer` has `t` and `d` reactive functions and `messages` reactive property to keep local messages.
for example, You can use the global `composer` to add localized messages to a component.
about detail  `composer`, see the API docs.
https://github.com/intlify/vue-i18n-next/blob/master/docs/vue-i18n.composer.md

**Comment by janein** [comment]
@houssemFat I just ran into the same issue and wondered if that's really the way to go. It just feels wrong to do the same thing in every single component which was a single line for the whole application in vue 2.

@kazupon I guess there is no "cleaner" or smarter way to handle this? In this case I think it would be nice to explicitly write this down in the docs. Was a litle bit confusing for me - I was thinking I was doing something wrong until I found this issue.
